Marille.Tests/CancellationTests.cs

```diff
@@ -154,4 +154,40 @@ public async Task MultithreadedClose ()
 		}
 		Assert.False (finalResult);
 	}
+
+	[Fact]
+	public async Task CloseAllChannelsAsync ()
+	{
+		// build several channels and then close them all, this should ensure that all the workers
+		// have consume all the messages
+		var eventCount = 100;
+		var list = new List<Task> (200);
+
+		configuration.Mode = ChannelDeliveryMode.AtLeastOnce;
+		var topic1 = "topic1";
+		var tcs1 = new TaskCompletionSource<bool> ();
+		var worker1 = new BlockingWorker(tcs1);
+		await _hub.CreateAsync<WorkQueuesEvent> (topic1, configuration);
+		await _hub.RegisterAsync (topic1, worker1);
+		
+		var topic2 = "topic2";
+		var tcs2 = new TaskCompletionSource<bool> ();
+		var worker2 = new BlockingWorker(tcs2);
+		await _hub.CreateAsync<WorkQueuesEvent> (topic2, configuration);
+		await _hub.RegisterAsync (topic2, worker2);
+		
+		for (var index = 0; index < eventCount; index++) {
+			await _hub.Publish (topic1, new WorkQueuesEvent($"myID{index}"));
+			await _hub.Publish (topic2, new WorkQueuesEvent($"myID{index}"));
+		}
+		
+		// we are blocking the consume of the channels
+		tcs1.SetResult (true);
+		tcs2.SetResult (true);
+
+		// close the hub, should throw no cancellation token exceptions and events should have been processed
```

Comment:
Can we assert that the tcs1 and tcs2 are not canceled?

Reply:
Done, TrySetResult should do the trick within an assert.